test/structure/test_chain.py

```diff
@@ -24,151 +23,130 @@ def test_sequence():
     assert chain.is_empty()
-    assert not chain.residues()
+    assert not chain.residues
 
 
 # TODO(CJ): need to add tests for checking if the chain has ceratin residues/residue types.
 
-@pytest.mark.TODO
+
 def test_same_sequence_equal():
     """Ensuring the Chain.is_same_sequence() method works for equivalent chains."""
     residues = [
-        Residue("A.A.1", list()),
-        Residue("A.A.2", list()),
-        Residue("A.B.3", list()),
-        Residue("A.D.5", list()),
+        Residue(1, "A", list()),
+        Residue(2, "A", list()),
+        Residue(3, "B", list()),
+        Residue(5, "D", list()),
     ]
     chain1 = Chain("A", residues)
     chain2 = Chain("A", residues)
     assert chain1.is_same_sequence(chain2)
     assert chain2.is_same_sequence(chain1)
 
     residues_alt = [
-        Residue("B.A.1", list()),
-        Residue("B.A.2", list()),
-        Residue("B.B.3", list()),
-        Residue("B.D.5", list()),
+        Residue(1, "A", list()),
+        Residue(2, "A", list()),
+        Residue(3, "B", list()),
+        Residue(5, "D", list()),
     ]
     chain3 = Chain("A", residues_alt)
     assert chain3.is_same_sequence(chain2)
     assert chain2.is_same_sequence(chain3)
 
+
 @pytest.mark.TODO
 def test_same_sequence_not_equal():
     """Ensuring the Chain.is_same_sequence() method works for non-equivalent chains."""
     residues = [
-        Residue("A.A.1", list()),
-        Residue("A.A.2", list()),
-        Residue("A.B.3", list()),
-        Residue("A.D.5", list()),
+        Residue(1, "A", list()),
+        Residue(2, "A", list()),
+        Residue(3, "B", list()),
+        Residue(5, "D", list()),
     ]
 
     chain1 = Chain("A", residues)
-    chain2 = Chain("B", residues)
-    assert not chain1.is_same_sequence(chain2)
-    assert not chain2.is_same_sequence(chain1)
 
     residues_alt1 = [
-        Residue("A.B.1", list()),
-        Residue("A.A.2", list()),
-        Residue("A.B.3", list()),
-        Residue("A.D.5", list()),
+        Residue(1, "B", list()),
+        Residue(2, "A", list()),
+        Residue(3, "B", list()),
+        Residue(5, "D", list()),
     ]
     chain3 = Chain("A", residues_alt1)
     assert not chain3.is_same_sequence(chain1)
     assert not chain3.is_same_sequence(chain1)
 
-    residues_alt2 = [
-        Residue("A.A.3", list()),
-        Residue("A.A.2", list()),
-        Residue("A.B.3", list()),
-        Residue("A.D.5", list()),
-    ]
-    chain4 = Chain("A", residues_alt2)
-    assert not chain4.is_same_sequence(chain1)
-    assert not chain1.is_same_sequence(chain4)
-
-    residues.pop()
-    residues.pop()
-    chain5 = Chain("A", residues)
+    rr = deepcopy(residues)
+    rr.pop()
+    rr.pop()
+    chain5 = Chain("A", rr)
     assert not chain5.is_same_sequence(chain1)
     assert not chain1.is_same_sequence(chain5)
 
-@pytest.mark.TODO
+
 def test_rename():
     """Testing the Chain.rename() method that gives the Chain() a new name."""
```

Comment:
Change name and docstring of the test to match the content

Reply:
Fixed.